lib/concurrent/executor/ruby_thread_pool_executor.rb

```diff
@@ -273,6 +276,18 @@ def ns_prune_pool
       @next_gc_time = Concurrent.monotonic_time + @gc_interval
     end
 
+    def ns_reset_if_forked
```

Comment:
`ns_shutdown_execution` should also call this, to ensure the pool is "valid" before it waits on it